cartridge/roles/metrics.lua

```diff
@@ -116,49 +108,78 @@ local function apply_routes(export)
         end
         if not is_present then
             delete_route(httpd, path)
-            current_paths[path] = nil
+            metrics_vars.current_paths[path] = nil
         end
     end
 end
 
-local metrics_config_present = false
-
--- removes routes that changed in config and adds new routes
-local function apply_config(conf)
-    local metrics_conf = conf.metrics
-    -- if metrics is not present in config then skip reconfiguring routes
-    if metrics_conf == nil then
+local function apply_default(paths)
```

Comment:
I'd suggest to leave `apply_routes` function alone, and call it with a config already merged.